src/context/Context.tsx

```diff
@@ -59,48 +36,68 @@ function MoviesShowsProvider({ children }: ContextProps) {
     // The search is performed only when there are 3 or more characters in the search bar
     // It should be triggered only one second after the user has stopped typing
     if (queryType === QUERY_TYPE.SEARCH) {
-      timer = setTimeout(
-        () =>
-          getItems(queryType).then(() => {
-            if (timer) {
-              clearTimeout(timer);
-              timer = null;
-            }
-          }),
-        DELAY,
-      );
+      timer = setTimeout(() => getItemsDataAndClearTimer(queryType), DELAY);
```

Comment:
As a new ticket investigate `debounce` and `throttle`, where to use them, and implement one of them as a helper function which you'll be using here.

Reply:
I created a new issue for this (#30).